src/legacy/core_plugins/embeddable_api/public/np_ready/public/lib/panel/panel_header/panel_actions/change_view_action.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { i18n } from '@kbn/i18n';
+import { IAction } from 'src/plugins/ui_actions/public';
+import { ContainerInput, IContainer } from '../../../containers';
+import { ViewMode, GetEmbeddableFactory, GetEmbeddableFactories } from '../../../types';
+import { NotificationsStart } from '../../../../../../../../../../core/public';
+import { KibanaReactOverlays } from '../../../../../../../../../../plugins/kibana_react/public';
+import { openChangeViewFlyout } from '../../../panel/panel_header/panel_actions/add_panel/open_change_view_flyout';
+
+export const CHANGE_VIEW_ACTION = 'changeView';
+
+interface ExpandedPanelInput extends ContainerInput {
+  expandedPanelId: string;
+}
+
+interface ActionContext {
+  embeddable: IContainer;
+}
+
+function hasExpandedPanelInput(
+  container: IContainer
+): container is IContainer<{}, ExpandedPanelInput> {
+  return (container as IContainer<{}, ExpandedPanelInput>).getInput().expandedPanelId !== undefined;
+}
+
+export class ChangeViewAction implements IAction<ActionContext> {
+  public readonly type = CHANGE_VIEW_ACTION;
+  public readonly id = CHANGE_VIEW_ACTION;
+  public order = 11;
+
+  constructor(
```

Comment:
nit: I think the pattern of taking in an object is easier to follow than a list of parameters, then order doesn't matter.

Reply:
Order influences the position of the new item ("Replace visualization") in the contextual menu.
I don't understand fully what you mean here.

Reply:
Sorry I meant order of the parameters, not the parameter named `order`. :)   When passing in a lot of arguments, a reader can know what they are, and a caller doesn't need to worry about which parameter is first, second third.  E.g. they can do :

```
new ChangeViewAction({ getFactory, getAllFactories });
```
or
```
new ChangeViewAction({ getAllFactories, getFactory});
```

So in situations like this, I find a single object being passed into the constructor easier to read.  It's just a nit though, feel free to leave as is if you like it this way better.

Reply:
I understand what you're saying, but I'm just copying the code that is present somewhere else, for example in add_panel_action.ts.

Reply:
ah, which I wrote, so I guess I don't follow my own advice, whoops! :D  Fine to leave as is.